pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -31,9 +31,15 @@ import (
 	"istio.io/istio/pkg/config/constants"
 	"istio.io/istio/security/pkg/k8s/chiron"
 	"istio.io/istio/security/pkg/pki/ca"
+	"istio.io/istio/security/pkg/pki/util"
+	"istio.io/pkg/env"
 	"istio.io/pkg/log"
 )
 
+var istiodTLSRSAKeySize = env.RegisterIntVar("ISTIOD_TLS_RSA_KEY_SIZE", 2048,
```

Comment:
I am not sure this is the right API for this. We had some proposals to add a first class API for CA related things we should sync up on. @shankgan @jacob-delgado do you know the status?

Reply:
I am not tracking this directly. The cert configs were a part of @jacob-delgado 's spec. Any deadline to merge them in?

Reply:
@jacob-delgado Would you please kindly take a look when you get a chance？Thanks！

Reply:
It got held up while I was resolving a few items. I'm fine with this change for now. There's currently `CITADEL_SELF_SIGNED_CA_RSA_KEY_SIZE` for `istiod` so this would be a nice addition. There was some debate on whether we should allow users to configure key sizes in that RFC as it could impact performance in that api, but as an environment variable it seems fine.

Is this going to be backported to 1.10? If I can get back to this for 1.11 (not sure, working on dual stack) I can add it to the list of env to be migrated.

Reply:
Thanks for the comments! I don’t have strong opinions on whether to be backported. @howardjohn PTAL, any comment or concern?

Reply:
To be perfectly honest, I am not convinced this should be merged at all. I am not an expert in this area and others seem to think it should be so that is fine. But regardless, I don't think we should backport a new feature unless there is a *very strong* user demand